### PR TITLE
Add subsystem to locally issue signed JWTs, replacing TokenRelay filter

### PR DESCRIPTION
**Pull request by vierbergenlars**
The new release of contentgrid-spring adds spring-oauth2-resource-server to the classpath which enforces by default authentication on all endpoints.

In a production scenario, the ContentGrid application runs behind the ContentGrid gateway, and receives a JWT (containing the user information) from the gateway.

Instead of directly forwarding the received authorization token (or the id token in case of an OIDC login in the gateway), the gateway will now generate a custom JWT to send to the application.

The JWT between gateway and application can also contain the Thunx ABAC context (so that one is signed as well)
For anonymous users (on the gateway) that are allowed; the gateway will also send a JWT, encoding the anonymous user information
For automation extension access tokens (when we add that); the gateway will also re-encode them, so user information is put in the token in a way that is consistent with normal user authentication.

### CODE REVIEW

File: src/main/java/com/contentgrid/gateway/security/jwt/issuer/LocallyIssuedJwtGatewayFilterFactory.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                JwtClaimsResolver.empty();
+
+        var signer = signerLocator.apply(config.getSigner());
+
+        Assert.notNull(signer, "No signer found with name %s".formatted(config.getSigner()));
+
+        return new LocallyIssuedJwtGatewayFilter(new SignedJwtIssuer(
+                signer,
+                claimsResolver
+        )) {
```
Comment (tgeens) [inline src/main/java/com/contentgrid/gateway/security/jwt/issuer/LocallyIssuedJwtGatewayFilterFactory.java:45]: why the new lines ?

File: src/test/java/com/contentgrid/gateway/security/jwt/issuer/PropertiesBasedJwtClaimsSignerTest.java [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+import lombok.Value;
+import lombok.experimental.Delegate;
+import org.junit.jupiter.api.Test;
+import org.springframework.core.io.support.ResourcePatternResolver;
+
+class PropertiesBasedJwtClaimsSignerTest {
+
+    static Random createDeterministicRandom() {
+        // This is a random that will always start from the same seed
+        return new Random(5);
```
Comment (tgeens) [inline src/test/java/com/contentgrid/gateway/security/jwt/issuer/PropertiesBasedJwtClaimsSignerTest.java:39]: I'd bet sonar is going to love this ;-)

File: src/main/java/com/contentgrid/gateway/security/jwt/issuer/PropertiesBasedJwtClaimsSigner.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+    private List<JWK> getActiveSigningKeys() {
+        return createSigningKeysFromPath(properties.getActiveKeys())
+                .toList();
+    }
+
+    private List<JWK> getAllSigningKeys() {
+        return Stream.concat(
+                getActiveSigningKeys().stream(),
+                createSigningKeysFromPath(properties.getAllKeys())
```
Comment (tgeens) [inline src/main/java/com/contentgrid/gateway/security/jwt/issuer/PropertiesBasedJwtClaimsSigner.java:88]: so properties.all-keys should not include the active-keys (or you would have duplicates here) ?
in that case, should we come up with a better name ?

File: src/test/java/com/contentgrid/gateway/security/jwt/issuer/PropertiesBasedJwtClaimsSignerTest.java (hunk 2 above)
Comment (vierbergenlars) [reply src/test/java/com/contentgrid/gateway/security/jwt/issuer/PropertiesBasedJwtClaimsSignerTest.java:39]: 1. This is a test class
2. Sonar doesn't seem to care about this one; it does complain about the other Random constructor in production code though.

File: src/main/java/com/contentgrid/gateway/security/jwt/issuer/PropertiesBasedJwtClaimsSigner.java (hunk 3 above)
Comment (vierbergenlars) [reply src/main/java/com/contentgrid/gateway/security/jwt/issuer/PropertiesBasedJwtClaimsSigner.java:88]: It is fine to have duplicates in here

**Review by tgeens — APPROVED** [review]

**Pull request merged**